src/__swaps__/screens/Swap/utils/assets.ts

```diff
@@ -0,0 +1,347 @@
+import { ChainId, ChainName } from '@/__swaps__/screens/Swap/types/chains';
+
+import { metadataClient } from '@/graphql';
+import * as i18n from '@/languages';
+import { SearchAsset } from '../types/search';
+
+import { chainIdFromChainName, chainNameFromChainId, customChainIdsToAssetNames, isNativeAsset } from './chains';
+import {
+  convertAmountAndPriceToNativeDisplay,
+  convertAmountToBalanceDisplay,
+  convertAmountToNativeDisplay,
+  convertAmountToPercentageDisplay,
+  convertAmountToRawAmount,
+  convertRawAmountToDecimalFormat,
+} from './numbers';
+import { Token } from '@/graphql/__generated__/metadata';
+
+const get24HrChange = (priceData?: ZerionAssetPrice) => {
+  const twentyFourHrChange = priceData?.relative_change_24h;
+  return twentyFourHrChange ? convertAmountToPercentageDisplay(twentyFourHrChange) : '';
+};
+
+export const getCustomChainIconUrl = (chainId: ChainId, address: AddressOrEth) => {
+  if (!chainId || !customChainIdsToAssetNames[chainId]) return '';
+  const baseUrl = 'https://raw.githubusercontent.com/rainbow-me/assets/master/blockchains/';
+
+  if (address === AddressZero || address === ETH_ADDRESS) {
+    return `${baseUrl}${customChainIdsToAssetNames[chainId]}/info/logo.png`;
+  } else {
+    return `${baseUrl}${customChainIdsToAssetNames[chainId]}/assets/${address}/logo.png`;
+  }
+};
+
+export const getNativeAssetPrice = ({ priceData, currency }: { priceData?: ZerionAssetPrice; currency: SupportedCurrencyKey }) => {
+  const priceUnit = priceData?.value;
+  return {
+    change: get24HrChange(priceData),
+    amount: priceUnit || 0,
+    display: convertAmountToNativeDisplay(priceUnit || 0, currency),
+  };
+};
+
+export const getNativeAssetBalance = ({
+  currency,
+  priceUnit,
+  value,
+}: {
+  currency: SupportedCurrencyKey;
+  decimals: number;
+  priceUnit: number;
+  value: string | number;
+}) => {
+  return convertAmountAndPriceToNativeDisplay(value, priceUnit, currency);
+};
+
+const isZerionAsset = (asset: ZerionAsset | AssetApiResponse): asset is ZerionAsset => 'implementations' in asset || !('networks' in asset);
+
+export function parseAsset({ asset, currency }: { asset: ZerionAsset | AssetApiResponse; currency: SupportedCurrencyKey }): ParsedAsset {
+  const address = asset.asset_code;
+  const chainName = asset.network ?? ChainName.mainnet;
+  const networks = 'networks' in asset ? asset.networks || {} : {};
+  const chainId = ('chain_id' in asset && asset.chain_id) || chainIdFromChainName(chainName) || Number(Object.keys(networks)[0]);
+
+  // ZerionAsset should be removed when we move fully away from websckets/refraction api
+  const mainnetAddress = isZerionAsset(asset)
+    ? asset.mainnet_address || asset.implementations?.[ChainName.mainnet]?.address || undefined
+    : networks[ChainId.mainnet]?.address;
+
+  const standard = 'interface' in asset ? asset.interface : undefined;
+
+  const uniqueId: UniqueId = `${mainnetAddress || address}_${chainId}`;
```

Comment:
if we do do this would love this in a helper so if we change for any reason its in 1 place vs multiple